patches/roslyn-analyzers/0001-WIP-for-source-build.patch

```diff
@@ -0,0 +1,1087 @@
++            {
++                var fileWithPath = Path.Combine(analyzerDocumentationFileDir, analyzerDocumentationFileName);
++                if (File.Exists(fileWithPath))
++                {
++                    result.AppendLine(FileElement(fileWithPath, "documentation"));
++                }
++            }
++
++            if (analyzerSarifFileDir.Length > 0 && Directory.Exists(analyzerSarifFileDir) && analyzerSarifFileName.Length > 0)
++            {
++                var fileWithPath = Path.Combine(analyzerSarifFileDir, analyzerSarifFileName);
++                if (File.Exists(fileWithPath))
++                {
++                    result.AppendLine(FileElement(fileWithPath, "documentation"));
++                }
++            }
++
++            if (analyzerConfigurationFileDir.Length > 0 && Directory.Exists(analyzerConfigurationFileDir) && analyzerConfigurationFileName.Length > 0)
++            {
++                var fileWithPath = Path.Combine(analyzerConfigurationFileDir, analyzerConfigurationFileName);
++                if (File.Exists(fileWithPath))
++                {
++                    result.AppendLine(FileElement(fileWithPath, "documentation"));
++                }
++            }
++
++            if (legacyRulesets.Length > 0)
++            {
++                foreach (string legacyRuleset in legacyRulesets)
++                {
++                    if (Path.GetExtension(legacyRuleset) == ".ruleset")
++                    {
++                        result.AppendLine(FileElement(Path.Combine(projectDir, legacyRuleset), @"rulesets\legacy"));
++                    }
++                }
++            }
++
++            result.AppendLine(FileElement(Path.Combine(assetsDir, "EULA.rtf"), ""));
++            result.AppendLine(FileElement(Path.Combine(assetsDir, "ThirdPartyNotices.rtf"), ""));
++            result.AppendLine(@"  </files>");
++
++            result.AppendLine(@"</package>");
++
++            File.WriteAllText(nuspecFile, result.ToString());
++
++        }
++
++    }
++}
+diff --git a/eng/common/dotnet-install.sh b/eng/common/dotnet-install.sh
+old mode 100644
+new mode 100755
+diff --git a/nuget/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.Package.csproj b/nuget/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.Package.csproj
+index 5ca2d53f9..c052f827c 100644
+--- a/nuget/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.Package.csproj
++++ b/nuget/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.Package.csproj
+@@ -14,7 +14,7 @@
+       PackageId is used by Restore. If we set it to Microsoft.CodeAnalysis.Analyzers 
+       Restore would conclude that there is a cyclic dependency between Microsoft.CodeAnalysis and Microsoft.CodeAnalysis.Analyzers.
+     -->
+-    <PackageId>*$(MSBuildProjectFullPath)*</PackageId>
++    <PackageId>*$(MSBuildProjectFullPath.Replace('/', '\\'))*</PackageId>
```

Comment:
As the comment in the original file says, they want to use the file path as the package name.

But when the "package name" (really the full file path with forward slashes on Linux) is put in the assets.json file, it ends up confusing the version parsing logic which thinks `"*/home/foo/bar/baz*/4.5.0"` has a version along the lines of `/home/foo/bar/baz*/4.5.0`.

Changing the slashes back to Windows-style slashes makes this work.

Reply:
Bizarre... can you add this to the code comment? (I imagine this can be upstreamed for compat, and would be nice to have explanation inline. If they don't want to find some other way.)

Reply:
Done.